dm/docs/RFCS/20220124_enhance_task_manageability.md

```diff
@@ -0,0 +1,127 @@
+# Proposal: Enhance task manageability
+
+- Author(s): [ehco](https://github.com/Ehco1996)
+- Last updated: 2022-01-22
+
+## Background
+
+The main purpose of this change is to address the root cause of problems like [#3771](https://github.com/pingcap/tiflow/issues/3771), which are caused by the fact that DM Task itself does not distinguish between dynamic configuration and static resources, making it impossible for users to intuitively manage their tasks. For this reasons, we will attempt to redesign the state machine of DM Task and optimising the DMCTL interaction interface to provide a better user experience.
+
+### Current State machine of task
+
+![dm old task lifecycle](../media/dm-old-task-lifecycle.png)
+
+### New State machine of task
+
+![dm new task lifecycle](../media/dm-new-task-lifecycle.png)
```

Comment:
"create" action should also begin from "null" state, or delete can go to black space.

"start" is placed in the line while other actions are off the line.

Reply:
> "start" is placed in the line while other actions are off the line.

dm-old-task-lifecycle.png also has this problem

Reply:
fixed now